### PR TITLE
facet range value docs

### PR DESCRIPTION
**Pull request by hiteshk25**
Ported following three commits
1. https://github.com/apache/lucene-solr/commit/bc0bce6c76f59162c3fe6cff64efdf81c0a7a4d1
2. https://github.com/apache/lucene-solr/commit/3e6915dfb12664893bbc2b0e7bb07a3b1ef119c1
3. https://github.com/apache/lucene-solr/commit/aa8485a235b05683d139c9b2f0a45e166d99a9ca

### CODE REVIEW

**Comment by claygoddard** [comment]
Was this a clean application? Looks like there are some build failures - are these related?

**Comment by hiteshk25** [comment]
> Was this a clean application? Looks like there are some build failures - are these related?

Lots of conflict but I haven't anything logical. Mostly formatting. Looks like many facet related tests are failing.
Looking further, we can run tests on this branch using gradle.

**Comment by hiteshk25** [comment]
@claygoddard here is one exception
```
2022-12-21T17:27:38.7650759Z   2>           => java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
2022-12-21T17:27:38.7651068Z   2> 	at org.apache.solr.search.facet.FacetRangeProcessor$IntCalc.toSortableDocValue(FacetRangeProcessor.java:1200)
2022-12-21T17:27:38.7651603Z   2> java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
2022-12-21T17:27:38.7651929Z   2> 	at org.apache.solr.search.facet.FacetRangeProcessor$IntCalc.toSortableDocValue(FacetRangeProcessor.java:1200) ~[main/:?]
2022-12-21T17:27:38.7652269Z   2> 	at org.apache.solr.search.facet.FacetRangeProcessor.createRangeList(FacetRangeProcessor.java:482) ~[main/:?]
2022-12-21T17:27:38.7652572Z   2> 	at org.apache.solr.search.facet.FacetRangeProcessor.process(FacetRangeProcessor.java:158) ~[main/:?]
2022-12-21T17:27:38.7652825Z   2> 	at org.apache.solr.search.facet.FacetRequest.process(FacetRequest.java:464) ~[main/:?]
2022-12-21T17:27:38.7653112Z   2> 	at org.apache.solr.search.facet.FacetProcessor.processSubs(FacetProcessor.java:524) ~[main/:?]
2022-12-21T17:27:38.7653390Z   2> 	at org.apache.solr.search.facet.FacetProcessor.fillBucket(FacetProcessor.java:469) ~[main/:?]
2022-12-21T17:27:38.7653668Z   2> 	at org.apache.solr.search.facet.FacetQueryProcessor.process(FacetQuery.java:60) ~[main/:?]
```

File: solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java [hunk 1]
```diff
@@ -818,6 +1193,12 @@ protected Integer parseStr(String rawval) {
     public Integer parseAndAddGap(@SuppressWarnings("rawtypes") Comparable value, String gap) {
       return ((Number) value).intValue() + Integer.parseInt(gap);
     }
+
+    @Override
+    @SuppressWarnings({"rawtypes"})
+    protected long toSortableDocValue(Comparable calcValue) {
+      return (long) calcValue;
+    }
```
Comment (claygoddard) [inline solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java:1201]: Your exception relates to this change here. Why did you change this to a direct cast?
The Solr 8 implementation had:
```
    @Override
    protected long toSortableDocValue(Comparable calcValue) {
      return ((Integer) calcValue).longValue();
    }
```

Comment (claygoddard) [reply solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java:1201]: Do you mind calling out any other changes you made for the patch?

Comment (hiteshk25) [reply solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java:1201]: Solr 9.0 gradle has some strict checking on type conversion task. We have couple of other similar tickets for type conversion. I will look this further.

File: versions.props [hunk 2]
```diff
@@ -13,7 +13,7 @@ com.google.guava:guava=31.1-jre
 com.google.re2j:re2j=1.2
 com.jayway.jsonpath:json-path=2.7.0
 com.lmax:disruptor=3.4.2
-com.tdunning:t-digest=3.1
+com.tdunning:t-digest=3.2
```
Comment (hiteshk25) [inline versions.props:16]: @magibney I have added your fix in this PR for now

**Review by claygoddard — APPROVED** [review]
LGTM

**Comment by hiteshk25** [comment]
One more test is failing like this; updating test for it
```
 3361837 INFO  (closeThreadPool-48495-thread-6) [] o.a.s.m.SolrMetricManager Closing metric reporters for registry=solr.jetty tag=null
   >     java.lang.AssertionError: expected:<483.70000000000016> but was:<486.5>
   >         at __randomizedtesting.SeedInfo.seed([7246FD61983861FD:FA12C2BB36C40C05]:0)
   >         at org.junit.Assert.fail(Assert.java:89)
   >         at org.junit.Assert.failNotEquals(Assert.java:835)
   >         at org.junit.Assert.assertEquals(Assert.java:555)
   >         at org.junit.Assert.assertEquals(Assert.java:685)
   >         at org.apache.solr.search.facet.DistributedFacetSimpleRefinementLongTailTest.checkSubFacetStats(DistributedFacetSimpleRefinementLongTailTest.java:517)
   >         at org.apache.solr.search.facet.DistributedFacetSimpleRefinementLongTailTest.checkSubFacetStats(DistributedFacetSimpleRefinementLongTailTest.java:462)
   >         at org.apache.solr.search.facet.DistributedFacetSimpleRefinementLongTailTest.test(DistributedFacetSimpleRefinementLongTailTest.java:82)
   >         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
   >         at org.apache.solr.BaseDistributedSearchTestCase$ShardsRepeatRule$ShardsFixedStatement.callStatement(BaseDistributedSearchTestCase.java:1166)
   >         at org.apache.solr.BaseDistributedSearchTestCase$ShardsRepeatRule$ShardsStatement.evaluate(BaseDistributedSearchTestCase.java:1137)
   >         at com.carrotsearch.randomizedtesting.rules.SystemPropertiesRestoreRule$1.evaluate(SystemPropertiesRestoreRule.java:80)
   >         at org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
   >         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
   >         at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
   >         at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
   >         at org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
   >         at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
   >         at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
   >         at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at com.carrotsearch.randomizedtesting.rules.SystemPropertiesRestoreRule$1.evaluate(SystemPropertiesRestoreRule.java:80)
   >         at org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
   >         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
   >         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
   >         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
   >         at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
   >         at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
   >         at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
   >         at org.junit.rules.RunRules.evaluate(RunRules.java:20)
   >         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   >         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
   >         at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
   >         at java.base/java.lang.Thread.run(Thread.java:829)
```
